pkg/csi/cephfsdeployment.go

```diff
@@ -35,183 +34,172 @@ var cephfsDeploymentLabels = map[string]string{
-			Name:      CephFSDeploymentName,
-			Namespace: namespace,
-			Labels:    cephfsDeploymentLabels,
-		},
-		Spec: appsv1.DeploymentSpec{
-			Replicas: ptr.To(int32(2)),
-			Selector: &metav1.LabelSelector{
-				MatchLabels: cephfsDeploymentLabels,
-			},
-			Template: corev1.PodTemplateSpec{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      CephFSDeploymentName,
-					Namespace: namespace,
-					Labels:    cephfsDeploymentLabels,
-				},
-				Spec: corev1.PodSpec{
-					ServiceAccountName: cephFSProvisionerServiceAccountName,
-					Containers: append([]corev1.Container{},
-						*provisioner,
-						*attacher,
-						*resizer,
-						*snapshotter,
-						v1.Container{
-							Name:            "csi-cephfsplugin",
-							Image:           sidecarImages.ContainerImages.CephCSIImageURL,
-							ImagePullPolicy: corev1.PullIfNotPresent,
-							Args: []string{
-								"--nodeid=$(NODE_ID)",
-								"--endpoint=$(CSI_ENDPOINT)",
-								"--v=5",
-								"--pidlimit=-1",
-								"--type=cephfs",
-								"--controllerserver=true",
-								fmt.Sprintf("--drivername=%s", GetCephFSDriverName()),
-							},
-							Env: []corev1.EnvVar{
-								{
-									Name: "POD_IP",
-									ValueFrom: &corev1.EnvVarSource{
-										FieldRef: &corev1.ObjectFieldSelector{
-											FieldPath: "status.podIP",
-										},
-									},
-								},
-								{
-									Name: "POD_NAMESPACE",
-									ValueFrom: &corev1.EnvVarSource{
-										FieldRef: &corev1.ObjectFieldSelector{
-											FieldPath: "metadata.namespace",
-										},
-									},
-								},
-								{
-									Name:  "CSI_ENDPOINT",
-									Value: templates.DefaultProvisionerSocketPath,
-								},
-							},
-							VolumeMounts: []corev1.VolumeMount{
-								{
-									Name:      "socket-dir",
-									MountPath: templates.DefaultSocketDir,
-								},
-								{
-									Name:      "host-dev",
-									MountPath: "/dev",
-								},
-								{
-									Name:      "host-sys",
-									MountPath: "/sys",
-								},
-								{
-									Name:      "lib-modules",
-									MountPath: "/lib/modules",
-									ReadOnly:  true,
-								},
-								{
-									Name:      "ceph-csi-configs",
-									MountPath: "/etc/ceph-csi-config",
-								},
-								{
-									Name:      "keys-tmp-dir",
-									MountPath: "/tmp/csi/keys",
+var cephFSDeploymentSpec = appsv1.DeploymentSpec{
+	Replicas: ptr.To(int32(2)),
+	Selector: &metav1.LabelSelector{
+		MatchLabels: cephfsDeploymentLabels,
+	},
+	Template: corev1.PodTemplateSpec{
+		Spec: corev1.PodSpec{
+			ServiceAccountName: cephFSProvisionerServiceAccountName,
+			Containers: []corev1.Container{
+				{Name: templates.ProvisionerContainer.Name},
+				{Name: templates.AttacherContainer.Name},
+				{Name: templates.ResizerContainer.Name},
+				{Name: templates.SnapshotterContainer.Name},
+				{
+					Name:            "csi-cephfsplugin",
+					Image:           sidecarImages.ContainerImages.CephCSIImageURL,
```

Comment:
Looks like you need to initialize `sidecarImages` in `csi.go` global scope.

Reply:
I think you are correct in your observation, but my solution will be to move setting of the image into the function that process the template copy

Reply:
Done